### PR TITLE
ssm-session-manager-plugin: add "aarch64-linux" support

### PR DESCRIPTION
**Pull request by WhyNotHugo**
###### Description of changes

This package has an `if` that covers only linux on amd64 and mac on all platforms. This adds support for linux of arm64.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage) **I am not using nixOS, so cannot run this**.
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

I created a flake that includes this package and it build and worked fine.

CC: @mbaillie

### CODE REVIEW

**Comment by WhyNotHugo** [comment]
@[martinbaillie](https://github.com/martinbaillie) I think you are the correct maintainer and the file has the wrong username?

**Comment by WhyNotHugo** [comment]
@maxeaubrey Maybe you're the right person to ping? :)

**Comment by maxeaubrey** [comment]
@WhyNotHugo looks good but I don't have an arm64 nix box to verify. I'm also not sure of the "correctness" of the way the different archs are gated (just visually comparing with other derivations in nixpkgs, not that there's not 50 different approaches 😅) - I'd suggest posting in the "PRs ready for review" thread.

**Comment by WhyNotHugo** [comment]
[This one](https://discourse.nixos.org/t/prs-ready-for-review/3032/1049)? Done.

**Review by martinbaillie — APPROVED** [review]

File: pkgs/applications/networking/cluster/ssm-session-manager-plugin/default.nix
Comment (SuperSandro2000) [inline pkgs/applications/networking/cluster/ssm-session-manager-plugin/default.nix:8]: Please use attr select instead of if else chains.

File: pkgs/applications/networking/cluster/ssm-session-manager-plugin/default.nix
Comment (SuperSandro2000) [inline pkgs/applications/networking/cluster/ssm-session-manager-plugin/default.nix:41]: There should not be a line break here while we are at it.

File: pkgs/applications/networking/cluster/ssm-session-manager-plugin/default.nix
Comment (WhyNotHugo) [reply pkgs/applications/networking/cluster/ssm-session-manager-plugin/default.nix:8]: I can't find any reference to `attr select` in the docs, and grepping `nixpkgs` doesn't yield anything useful.

Can you point me to an example of what you have in mind? I can use that as a reference.

Comment (WhyNotHugo) [reply pkgs/applications/networking/cluster/ssm-session-manager-plugin/default.nix:8]: Oh, I think [it's this](https://gist.github.com/joepie91/c3c047f3406aea9ec65eebce2ffd449d)?

I'm not sure how to write it for this particular case; there is not one branch per platform. There are four platforms, but three branches, since two share the same binaries.

Comment (SuperSandro2000) [reply pkgs/applications/networking/cluster/ssm-session-manager-plugin/default.nix:8]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/servers/prowlarr/default.nix#L12-L22

> There are four platforms, but three branches, since two share the same binaries.

This can be solved like

```nix
  hash = rec {
    x64-linux = "sha256-NTkb6E0GRofd63TozLPxyNZ6GxO4pAc+MWsUDKWLZDY=";
    arm64-linux = "sha256-eB1Izvf/E0iOGkWt6fSc4oAt1sJ7XTxn/iwDNYz7WkA=";
    x64-osx = "sha256-TAphdcbzbNUZpWYVaHBkhI+RsgxWmI8n8Z6VwYxHXWg=";
    arm64-osx = x64-osx;
  }."${arch}";
```

File: pkgs/applications/networking/cluster/ssm-session-manager-plugin/default.nix
Comment (SuperSandro2000) [inline pkgs/applications/networking/cluster/ssm-session-manager-plugin/default.nix:16]:
```suggestion
    }.$system or (throw "Unsupported system: ${system}");
```
forgot to mention that. Same for line 10. Also I am not sure if that works, yet. system seems to be undefined.

Comment (WhyNotHugo) [reply pkgs/applications/networking/cluster/ssm-session-manager-plugin/default.nix:16]: It's not: [`error: undefined variable 'system'`](git@github.com:pimutils/khal.git).

`arch` isn't enough here, it also depends on what OS we're building for.

Comment (SuperSandro2000) [reply pkgs/applications/networking/cluster/ssm-session-manager-plugin/default.nix:16]: I think you want to replace ``system`` with ``stdenv.hostPlatform.system``.

**Comment by WhyNotHugo** [comment]
CI seems happy now. I've tested building on aarch64 and amd64.

**Comment by SuperSandro2000** [comment]
@ofborg build ssm-session-manager-plugin

**Comment by SuperSandro2000** [comment]
Can you please fix the commit message to fit the contributing guide? Then ofborg would also be auto triggered.

After that we can merge this.

**Comment by WhyNotHugo** [comment]
> Can you please fix the commit message to fit the contributing guide?

https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#writing-good-commit-messages

I don't see any specific convention mentioned. History makes it obvious what I should use, but perhaps an explicit mention in the docs would help? I'm very sure my previous contributions don't fit this very clear convention.

**Comment by WhyNotHugo** [comment]
I think ofborg is happy now :relieved:

**Comment by SuperSandro2000** [comment]
First bullet in https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#submitting-changes

**Comment by WhyNotHugo** [comment]
Thanks!

**Comment by WhyNotHugo** [comment]
Thanks!

**Pull request merged**
